packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageModule.kt

```diff
@@ -96,7 +101,37 @@ class ExpoImageModule : Module() {
       }
     }
 
-    AsyncFunction<Boolean>("clearMemoryCache") {
+    AsyncFunction("loadAsync") { source: SourceMap, promise: Promise ->
+      CoroutineScope(Dispatchers.Main).launch {
+        ImageLoadTask(appContext, source).load(promise)
+      }
+    }
+
+    Class(Image::class) {
+      Property("width") { image: Image ->
+        image.ref?.intrinsicWidth
+      }
+      Property("height") { image: Image ->
+        image.ref?.intrinsicHeight
+      }
+      Property("scale") { image: Image ->
+        // Not relying on `2x` in the filename, but want to make the following true:
+        //  If you multiply the logical size of the image by this value, you get the dimensions of the image in pixels.
+        val screenDensity = appContext.reactContext?.resources?.displayMetrics?.density ?: 1f
+        (image.ref?.toBitmapOrNull()?.density ?: 1) / (screenDensity * 160.0f)
+      }
+      Property("isAnimated") { image: Image ->
+        if (image.ref is GifDrawable) {
+          return@Property true
+        }
+        false
+      }
+      Property("mediaType") { ->
+        null // not easily supported on Android https://github.com/bumptech/glide/issues/1378#issuecomment-236879983
```

Comment:
Can we at least support some of the common image types by inferring them from the file extension?

Reply:
We don't really have an extension in a Drawable. I could do that just for animated drawables, maybe @lukmccall knows a better way.

Reply:
We would have to create a custom drawable. We had similar problem decoding SVG - https://github.com/expo/expo/blob/12aeff5b52f9367ee519bd2996f0ea42bc15fcd9/packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGDrawableTranscoder.kt#L19 

Also, we're not passing media type in load event for the same reasons. It's a bigger problem.